src/Missings.jl

```diff
@@ -208,6 +208,128 @@ missing
 """
 passmissing(f) = PassMissing{Core.Typeof(f)}(f)
 
+struct SpreadMissings{F} <: Function
+    f::F
+end
+
+function (f::SpreadMissings{F})(args...; kwargs...) where {F}
+    kwargs_vals = values(values(kwargs))
+    xs = tuple(args..., kwargs_vals...)
+    if any(x -> x isa AbstractVector{>:Missing}, xs)
+        vecs = Base.filter(x -> x isa AbstractVector, xs)
+
+        findex = eachindex(first(vecs))
+        if !(all(x -> eachindex(x) == findex, vecs[2:end]))
+            d = Dict()
+            for i in 1:length(vecs)
+                e = eachindex(vecs[i])
+                if eachindex(e) in keys(d)
+                    push!(d[i], i)
+                else
+                    d[e] = [i]
+                end
+            end
+            s = "The indices of vector-input arguments are not all " *
+                "the same.\n"
+
+            for k in keys(d)
+                inds = join(d[k], ", ", " and ")
+                ind_msg = "Vector inputs $inds have indices $k \n"
+                s = s * ind_msg
+            end
+
+            throw(ArgumentError(s))
+        end
+
+        nonmissingmask = fill(true, length(vecs[1]))
+        for v in vecs
+            nonmissingmask .&= .!ismissing.(v)
```

Comment:
As I noted in an outdated comment, it would be faster to do `nonmissinginds = findall(nonmissingmask)` and use that below instead of `nonmissingmask`, which has to recompute integer indices each time it's used.

Reply:
Added.